Optically center the attachment download/save icon

squelart opened the pull request:
#### 6216d63544a99c31cf058d762bc4f547b241fc70
<pre>
Optically center the attachment download/save icon
<a href="https://bugs.webkit.org/show_bug.cgi?id=265221">https://bugs.webkit.org/show_bug.cgi?id=265221</a>
<a href="https://rdar.apple.com/116239561">rdar://116239561</a>

Reviewed by Tim Nguyen.

The icon is heavier at the bottom, so it needs to be placed slightly
higher than the center line to be perceived as optically centered.

* Source/WebCore/html/shadow/attachmentElementShadow.css:
(button#attachment-save-button::before):

Canonical link: <a href="https://commits.webkit.org/271037@main">https://commits.webkit.org/271037@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ca07214c82a30150f0cdf88a16373c9d6f051922

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/27152 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5768 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/28388 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/29360 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24826 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/27610 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/7672 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/3163 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24667 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/27414 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/4598 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/23302 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/4002 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/4121 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/24299 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29995 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24799 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24713 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/30291 "Found 1 new test failure: fast/events/monotonic-event-time.html (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/4117 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/2300 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/28212 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5579 "Built successfully") | | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/6540 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4576 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3514 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/4493 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/html/shadow/attachmentElementShadow.css:
```diff
@@ -201,21 +201,22 @@ button#attachment-save-button {
     mask-repeat: no-repeat;
-    mask-position: center;
+    mask-size: var(--mask-size);
+    mask-position: 50% calc(50% - (6/64) * var(--mask-size));
```
Can you add a comment that describes that this is to make things optically centered?

nt1m replied:
Potentially describing that 64 is the image natural size

squelart committed:
Optically center the attachment download/save icon
https://bugs.webkit.org/show_bug.cgi?id=265221
rdar://116239561

Reviewed by Tim Nguyen.

The icon is heavier at the bottom, so it needs to be placed slightly
higher than the center line to be perceived as optically centered.

* Source/WebCore/html/shadow/attachmentElementShadow.css:
(button#attachment-save-button::before):

Canonical link: https://commits.webkit.org/271037@main
